stats/stats_test.go

```diff
@@ -720,7 +720,50 @@ func TestNaN(t *testing.T) {
 	}
 }
 
+func TestBucketLookUp(t *testing.T) {
+	var tests = []struct {
+		input float64 // input
+		start float64 // start
+		end   float64 // end
+
+	}{
+		{input: 999, start: 900, end: 1000},
+		{input: 999.99, start: 900, end: 1000},
+		{input: 1000.0004, start: 1000, end: 2000},
+		{input: 1000.01, start: 1000, end: 2000},
+		{input: 1001, start: 1000, end: 2000},
+		{input: 1999.99, start: 1000, end: 2000},
+		{input: 2000.0004, start: 2000, end: 3000},
+		{input: 2001, start: 2000, end: 3000},
+	}
+	h := NewHistogram(0, 1)
+	for _, test := range tests {
+		h.Reset()
+		h.Record(1)
+		h.Record(1000000)
+		h.Record(test.input)
+		hData := h.Export()
+		if hData.Data[1].Start != test.start || hData.Data[1].End != test.end {
+			t.Errorf("Testing data %f is matched with different bucket", test.input)
```

Comment:
please %+v print tests and the actual value

in go tests always print the actual value first 

so... "Got %+v while expected %+v", hData.Data[1], test)